File: src/types/clipboard.ts
/**
 * Clipboard 数据类型 — 对应架构文档 §2.1 表4
 */

export interface ClipboardItem {
  id: number
  content_type: 'text' | 'code' | 'image'
  text_content: string | null
  image_path: string | null
  ocr_text: string | null
  category: string | null
  tags: string
  is_pinned: number
  content_hash: string
  created_at: string
}

File: src-tauri/src/models/clipboard.rs
use serde::Serialize;

/// 剪贴板历史（对应 clipboard_items 表）
#[derive(Debug, Clone, Serialize)]
pub struct ClipboardItem {
    pub id: i64,
    pub content_type: String,
    pub text_content: Option<String>,
    pub image_path: Option<String>,
    pub ocr_text: Option<String>,
    pub category: Option<String>,
    pub tags: String,
    pub is_pinned: i32,
    pub content_hash: String,
    pub created_at: String,
}
